### PR TITLE
Scala inference memory leak fix #11204

### PR DESCRIPTION
**Pull request by andrewfayres**
## Description ##
This fixes a memory leak that results from the FeedForward.predict method not properly disposing of the results from NDArray.slice().

### Verifying leak exists ###
To verify that there was a leak in the existing code I created and trained a basic model off of MNIST data then called the predict method inside a while(true) loop then monitored the process's memory usage.

### Verifying leak fix ###
After making the code changes, I repeated the process I had used to verify there was a leak. Memory consumption was much improved and looks to be stable.

### More testing ###
In addition to the tests I ran to monitor memory usage, I also ran the existing tests with 'make scalatest'. All tests pass. 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by andrewfayres** [comment]
@nswamy @lanking520

**Comment by lanking520** [comment]
Hi @marcoabreu , this issue has been found by a customer and we feel it really important to the Scala package. It will crash users inference server if we didn't fix that. The memory leak problem has been addressed by a warning in the system. It can be found during runtime.

**Comment by lanking520** [comment]
Hi @marcoabreu , we have a PR here: https://github.com/apache/incubator-mxnet/pull/11204

**Comment by marcoabreu** [comment]
Ah Excellent, thanks a lot. sorry for the inconvenience

File: scala-package/core/src/main/scala/org/apache/mxnet/FeedForward.scala [hunk 1]
```diff
@@ -224,13 +224,24 @@ class FeedForward private(
...
     while (data.hasNext && i != numBatch) {
       val batch = data.next()
-      i += 1
-      ExecutorManager.loadData(batch, dataArrays)
-      predExec.forward(isTrain = false)
-      val padded = batch.pad
-      val realSize = batchSize - padded
-      for ((list, nd) <- outputs zip predExec.outputs) {
-        list += nd.slice(0, realSize).copy()
+      try {
```
Comment (nswamy) [inline scala-package/core/src/main/scala/org/apache/mxnet/FeedForward.scala]: I find the try finally really annoying and redundant. MXNet actually seg-fualts for simple cases such as the input shape is wrong. We are adding these try-finally and making the code more complicated

**Pull request merged**